### PR TITLE
Feat/create cases for microbial samples

### PR DESCRIPTION
**Pull request by patrikgrenfeldt**
This PR adds cases for microbial samples

### How to prepare for test
- [x] ssh to hasta (depending on type of change)
- [x] install on stage:
`bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-cg-stage.sh [THIS-BRANCH-NAME]`

- [x] ssh to clinical-db (depending on type of change)
- [x] install on stage:
`bash /home/proj/production/servers/resources/clinical-db.scilifelab.se/update-clinical-api-stage.sh [THIS-BRANCH-NAME]`
- [x] take fresh dbcopy to stage of cg from production

### How to test creating cases for existing microbial samples
- [x] login to hasta
- [x] run script cg/scripts/create_cases_for_microbial_orders.py

### Expected test outcome
- [x] check that you can't find any samples that have data_analysis like microbial that lacks a case

### How to test that ordering creates new cases
- [x] login to orderportal
- [x] submit an microbial order for a ticket that has not been used before

### Expected test outcome
- [x] check in the family view in clinical-api-atage that a new case has been created

### How to test that ordering adds to existing cases
- [x] login to orderportal
- [x] submit an microbial order for a ticket that has been used before

### Expected test outcome
- [x] check in the family view in clinical-api-atage that the samples was added to an existing case

- [x] Take a screenshot and attach or copy/paste the output.

## Review
- [x] code approved by MM
- [x] tests executed by PG
- [x] "Merge and deploy" approved by PG
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: cg/meta/orders/status.py [hunk 1]
```diff
@@ -393,8 +400,12 @@ def store_microbial_samples(
                     ticket=ticket,
                 )
 
+                priority = new_sample.priority
```
Comment (moonso) [inline cg/meta/orders/status.py:403]: Looks like something is not really thought through when using this many arguments in a function call, right? Could this be split into several functions? Could we use some kind of data class to hold all this values and pass objects of that class?

File: tests/meta/orders/test_meta_orders_status.py [hunk 2]
```diff
@@ -215,7 +215,7 @@ def test_store_microbial_samples(orders_api, base_store, microbial_status_data):
 
     # THEN it should store the samples and the used previously unknown organisms
     assert new_samples
-    assert base_store.families().count() == 0
+    assert base_store.families().count() == 1
```
Comment (moonso) [inline tests/meta/orders/test_meta_orders_status.py:218]: Could you add "THEN" statements for all asserts? It is hard to understand why the test has changed without any explanation

File: tests/meta/orders/test_meta_orders_status.py [hunk 3]
```diff
@@ -240,7 +240,7 @@ def test_store_microbial_samples_data_analysis_stored(
 
     # THEN it should store the samples
     assert base_store.samples().count() > 0
-    assert base_store.families().count() == 0
+    assert base_store.families().count() > 0
```
Comment (moonso) [inline tests/meta/orders/test_meta_orders_status.py:243]: Same here. Why was it 0 before and now one? Needs commenting

**Review by moonso — CHANGES_REQUESTED** [review]
Nice @patrikgrenfeldt ! I had some comments

File: cg/meta/orders/status.py (hunk 1 above)
Comment (patrikgrenfeldt) [reply cg/meta/orders/status.py:403]: I think you are right though I don't want to do that now, this was just meant to be a small PR to allow Barry to continue his work on storing microbial analysis results for the swedac deviation. Let's save that for later.

File: tests/meta/orders/test_meta_orders_status.py (hunk 2 above)
Comment (patrikgrenfeldt) [reply tests/meta/orders/test_meta_orders_status.py:218]: fixed

File: tests/meta/orders/test_meta_orders_status.py (hunk 3 above)
Comment (patrikgrenfeldt) [reply tests/meta/orders/test_meta_orders_status.py:243]: fixed

**Comment by patrikgrenfeldt** [comment]
Test result creating cases for all samples:
image
👍

**Review by moonso — APPROVED** [review]

**Comment by patrikgrenfeldt** [comment]
Test result order without existing family: 
image

Family created: 👍

**Comment by patrikgrenfeldt** [comment]
Test result adding to family: 
DB Before first order:
image
DB after first order:
image
Family after first order:
image

Family after second order: 


👍

**Pull request merged**